### PR TITLE
Prototype azure-storage-queue using azure-xml

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Prototype of `azure-storage-queue` using `azure-xml` for handling XML serialization instead of Jackson Dataformat XML.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/implementation/models/ListQueuesSegmentResponse.java [hunk 1]
```diff
@@ -188,4 +175,94 @@ public ListQueuesSegmentResponse setNextMarker(String nextMarker) {
...
+    /**
+     * Reads an instance of ListQueuesSegmentResponse from the XmlReader.
+     *
+     * @param xmlReader The XmlReader being read.
+     * @return An instance of ListQueuesSegmentResponse if the XmlReader was pointing to an instance of it, or null if
+     *     it was pointing to XML null.
+     * @throws IllegalStateException If the deserialized XML object was missing any required properties.
+     * @throws XMLStreamException If an error occurs while reading the ListQueuesSegmentResponse.
+     */
+    public static ListQueuesSegmentResponse fromXml(XmlReader xmlReader) throws XMLStreamException {
```
Comment (srnagar) [inline sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/implementation/models/ListQueuesSegmentResponse.java:212]: Are these `fromXml` and `toXml` methods generated?

File: sdk/storage/azure-storage-queue/src/main/java/module-info.java [hunk 2]
```diff
@@ -2,21 +2,16 @@
 // Licensed under the MIT License.
 
 module com.azure.storage.queue {
+    requires transitive com.azure.xml;
```
Comment (srnagar) [inline sdk/storage/azure-storage-queue/src/main/java/module-info.java:5]: Do we need to make this transitive? Do we expect customers to use any of the Xml serializer APIs from azure-xml directly?

File: sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/implementation/models/ListQueuesSegmentResponse.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/implementation/models/ListQueuesSegmentResponse.java:211]: Yeah, these are generated

File: sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java [hunk 3]
```diff
@@ -1289,12 +1286,8 @@ PagedFlux<PeekedMessageItem> peekMessagesWithOptionalTimeout(Integer maxMessages
     }
 
     private Mono<PagedResponseBase<MessagesPeekHeaders, PeekedMessageItem>> transformMessagesPeekResponse(
-        ResponseBase<MessagesPeekHeaders, List<PeekedMessageItemInternal>> response) {
-        List<PeekedMessageItemInternal> peekedMessageInternalItems = response.getValue();
-        if (peekedMessageInternalItems == null) {
```
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java:1294]: I'll have an update changing this soon

**Pull request closed without merging**